File: 공부내용정리/알고리즘/programmers/HashMap/PhoneBookList/PhoneBookList.ts
function solution(phone_book) {
    
  const phoneBookSet = new Set(phone_book);
  
  for (const number of phone_book) {
      for ( let i = 1; i < number.length; i++) {
          
          const prefix = number.substring(0, i);
          
          if ( phoneBookSet.has(prefix) ) {
              return false;
          }
      }
  }
  return true;
}

File: 공부내용정리/알고리즘/programmers/HashMap/PhoneBookList/PhoneBookList.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        
        Map<String, Boolean> phoneBookMap = new HashMap<>();
        
        for (String number : phone_book) {
            phoneBookMap.put(number, true);
        }
        
        for (String number: phone_book) {
            for (int i = 1; i < number.length(); i++ ) {
                String prefix = number.substring(0, i);
                if ( phoneBookMap.containsKey(prefix) ) {
                    return false;
                }
            }
        }
        
        return true;
    }
}